docs/api-guide.md

```diff
@@ -786,107 +786,115 @@ Let's look at the following example code to see how to handle exceptions in Scal
+If you catch `TransactionException`, it indicates that the transaction fails to begin due to transient or nontransient faults. You can try retrying the transaction, but you may not be able to begin the transaction due to nontransient faults.
+If you catch `TransactionNotFoundException`, it indicates that the transaction fails to begin due to transient faults. You can retry the transaction.
 
 The APIs for CRUD operations (`get()`/`scan()`/`put()`/`delete()`/`mutate()`) could throw `CrudException` and `CrudConflictException`.
-If you catch `CrudException`, it indicates some failure (e.g., database failure and network error) happens during the transaction, so you should cancel the transaction or retry the transaction after the failure or error is fixed.
-If you catch `CrudConflictException`, it indicates a transaction conflict occurs during the transaction, so you can retry the transaction from the beginning, preferably with well-adjusted exponential backoff based on your application and environment.
-The sample code retries three times maximum and sleeps 100 milliseconds before retrying the transaction.
+If you catch `CrudException`, it indicates that the transaction CRUD operation fails due to transient or nontransient faults. You can try retrying the transaction from the beginning, but the transaction may still fail if the cause is nontranient.
+If you catch `CrudConflictException`, it indicates that the transaction CRUD operation fails due to transient faults (e.g., a conflict error). You can retry the transaction from the beginning. 
+
+The APIs for mutation operations (`put()`/`delete()`/`mutate()`) could also throw `UnsatisfiedConditionException`.
+If you can this exception, it indicates that the condition for the mutation operation is not met.
+You can handle this exception according to your application requirements.
 
 Also, the `commit()` API could throw `CommitException`, `CommitConflictException`, and `UnknownTransactionStatusException`.
-If you catch `CommitException`, like the `CrudException` case, you should cancel the transaction or retry the transaction after the failure or error is fixed.
-If you catch `CommitConflictException`, like the `CrudConflictException` case, you can retry the transaction from the beginning.
-If you catch `UnknownTransactionStatusException`, you are not sure if the transaction succeeds or not.
+If you catch `CommitException`, it indicates that committing the transaction fails due to transient or nontransient faults. You can try retrying the transaction from the beginning, but the transaction may still fail if the cause is nontranient.
+If you catch `CommitConflictException`, it indicates that committing the transaction fails due to transient faults (e.g., a conflict error). You can retry the transaction from the beginning.
+If you catch `UnknownTransactionStatusException`, it indicates that you are not sure if the transaction succeeds or not.
```

Comment:
```suggestion
If you catch `UnknownTransactionStatusException`, this indicates that you are not sure if the transaction succeeds or not.
```

Although I edited the above sentence a bit, it is a bit unclear to me. Should `this indicates that you are not sure if the transaction succeeds or not` be something like `this indicates that the status of the transaction, whether it has succeeded or not, is unknown` instead?

Reply:
Sounds good. Thank you! Fixed in 0b33081cfbc4f37516948c8d6c4b84821c2f0a29.